### ISSUE TITLE
Boolean values not evaluated as strings

### ISSUE THREAD

**Issue by getodk-bot** [getodk/javarosa#558]
<a href="https://github.com/mitchellsundt"><img src="https://avatars.githubusercontent.com/u/2281171?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [mitchellsundt](https://github.com/mitchellsundt)**
_Thursday Jul 09, 2015 at 19:46 GMT_
_Originally opened as https://github.com/opendatakit/opendatakit/issues/1136 (0 comment(s))_

---

Originally reported on Google Code with ID 1135

```
Originally reported as a bug with Enketo; Enketo thinks their behavior is spec-compliant.

https://github.com/kobotoolbox/enketo-express/issues/193

Basically, if f is false(), then ODK Collect should interpret (${f} = "false") as true.
```

Reported by `nils@wlonk.com` on 2015-04-24 20:51:21

**Comment by macdude357** [comment]
I looked into this.  The existing logic follows the XPath conversion rules which state that if the length of the String is > 0, then then True is returned, otherwise False is returned.  Unfortunately, this is different from the rules that Java uses.  So, it appears that javarosa is behaving properly according to the XPath spec.

**Comment by lognaturel** [comment]
Thanks so much for all the triage, tests and fixes, @macdude357! 🥇 

I think I've already mentioned this to you but consider hopping into [Slack](http://slack.getodk.org/) if you ever want to discuss anything you're doing synchronously.
